core/modules/admin_bar/admin_bar.module

```diff
@@ -244,6 +244,27 @@ function admin_bar_preprocess_page(&$variables) {
     $settings['position_fixed'] = $setting;
   }
 
+  // Send whether this is an admin page to JS for back to site functionality.
+  $settings['current_path_is_admin'] = FALSE;
+  $settings['current_path'] = current_path();
+  if (path_is_admin($settings['current_path'])) {
+    $settings['current_path_is_admin'] = TRUE;
+  }
+  /*$settings['back_to_site_link'] = config_get('admin_bar.settings',
+   'back_to_site_link');*/
+
+  //$settings['back_to_site_link'] = config_get('admin_bar.settings',
+  // 'components . [admin_bar.back_to_site_link]');
+
+  $components = config_get('admin_bar.settings', 'components');
```

Comment:
I suggest replacing with:

```php
$components = $config->get('components');
```

(watch your spaces between closing parentheses in the following line).